log4j-core/src/main/java/org/apache/logging/log4j/core/config/Property.java

```diff
@@ -118,15 +119,36 @@ public static Property createProperty(final String name, final String value) {
      * @param value The value.
      * @return A Property.
      */
+    public static Property createProperty(
+            final String name,
+            final String rawValue,
+            final String value) {
+        if (name == null) {
+            LOGGER.error("Property name cannot be null");
+        }
+        return new Property(name, rawValue, value);
+    }
+
+    /**
+     * Creates a Property.
+     *
+     * @param name The key.
+     * @param rawValue The value without any substitution applied.
+     * @param configuration configuration used to resolve the property value from the rawValue
+     * @return A Property.
+     */
     @PluginFactory
     public static Property createProperty(
             @PluginAttribute("name") final String name,
             @PluginValue(value = "value", substitute = false) final String rawValue,
-            @PluginValue("value") final String value) {
+            @PluginConfiguration final Configuration configuration) {
```

Comment:
Attribute are consumed as they're read, this way we consume the value once, and process it using the configuration substitutor.